src/react-vite/01-intro-to-react/intro-to-react.md

```diff
@@ -8,45 +8,41 @@
 
 ## Overview
 
-TODO
-
-## Objective 1
-
-TODO
-
-### Key concepts
-
-TODO
-
-#### Concept 1
-
-TODO
-
-#### Concept 2
-
-TODO
-
-### Requirements
-
-TODO
-
-### Setup
-
-TODO
-
-### Verify
-
-TODO
-
-### Solution
-
-<details>
-<summary>Click to see the solution</summary>
-
-TODO
-
-</details>
-
-## Next steps
-
-TODO
\ No newline at end of file
+React first appeared in 2013 with "components" as the basic building block for
+constructing a user interface. Initially components were created using
+JavaScript classes which implemented an interface that defined lifecycle
+methods. One of React's notable characteristics has existed from the very
+beginning: the use of [JSX
+syntax](https://react.dev/learn#writing-markup-with-jsx) to define the structure
+of a component.
+
+### Key React Concepts
+
+- Data in a React based application flows one way: into a component through
+`props` or state then the user interface is rendered, user interactions cause
+changes to state, changes to state cause props to be updated and passed to the
+component again - restarting the cycle.
+
+- React's Virtual DOM allows React to quickly determine when the user interface
+needs to be updated; since updates to the browser's DOM are time-consuming only
+making them when necessary is critical to React's rendering performance. When
+the state provided to a React component changes the React framework will invoke
```

Comment:
I would run this text through a grammar checker because it should catch things like `component changes the React` (should have a comma, i.e. `component changes, the React`).

Reply:
Done